### ISSUE TITLE
Tooltips are not announced by screen readers

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3525]
## Describe the bug

Tooltips are not announced when you tab and focus on them.

## How to reproduce

https://clarity.design/documentation/tooltips

- Using a screen reader, tab to any of the tooltip links and observe that you do not hear the tooltip text when it appears.

## Expected behavior

_A clear and concise description of what you expected to happen._

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

This is due the the tooltip being inside the link that triggers and not using aria-describedby.

Separate the tooltip from the link and use aria-describedby to use the tooltip to describe the link.

See the following:
https://github.com/clane/Angular-A11y-Examples/blob/master/src/app/tootip/tootip.component.html

and 

http://w3c.github.io/aria-practices/#tooltip

**Comment by mathisscott** [comment]
@chlane 
Please see included screen recording. VoiceOver is reading tooltips just fine. What am I missing?

[ScreenReaderReadsTooltips.mov.zip](https://github.com/vmware/clarity/files/3374213/ScreenReaderReadsTooltips.mov.zip)

**Comment by chlane** [comment]
@mathisscott 
The issue can be reproduced with MacOS/Safari/VoiceOver and Windows 10/JAWS 2019/IE 11. 

I recommend we follow the implementation at 

http://whatsock.com/tsg/Coding%20Arena/Tooltips/Tooltip%20(Internal%20Content)/demo.htm#

which uses aria-describedby with the following structure:

<button id="tooltipId" href="#" class="accTooltip helpIcon" data-src="files/tooltip.html#tooltip-help" data-offsetleft="-200" data-offsettop="70" aria-describedby="AccDC1562702295705">
                	What is the greatest secret known to man? <img src="img/tooltip/ic_help.svg" alt="Help"></button>

<div id="AccDC1562702783594" class="tooltip" role="region" aria-label="tooltip" style="position: absolute; left: 627px; top: 943px;"><div id="AccDC1562702783778" tabindex="-1" role="tooltip"><div><div id="tooltip-help">
"It is a mistake to think you can solve any major problems just with potatoes.", Douglas Adams
</div></div></div></div>"

**Comment by mathisscott** [comment]
From digging into the guts of that tooltip, it looks like we want to change our HTML to look more like...

```
<span aria-describedby="user-gives-me-an-Id">I am describing this thing</span>
<a href="#" role="region" aria-label="tooltip" class="tooltip tooltip-sm tooltip-top-right">
    <clr-icon shape="info-circle" size="24" title="Lorem ipsum sit"></clr-icon>
    <span id="user-gives-me-an-Id" class="tooltip-content" tabindex="-1" role="tooltip">Lorem ipsum sit</span>
</a>
```
As I see it, this would be a non-breaking change that could be backported – largely adding static attributes to the `clr-tooltip` and also an input to force the user-provided id. If no id is provided... 🤔 

Note the following comments in that code: 
```
When an A tag is used as the triggering element, it must adhere to the following:
1. If the element contains no link text and no Img element, then the link should include an informative Title attribute. (This is only valid if there is no link text at all)
2. If the element contains an Img tag and no link text, then the Img tag should include an informative Alt attribute and Title attribute, both of which should match.
3. The A tag must include an Href attribute, to ensure keyboard accessibility.

This will ensure the highest level of accessibility for both screen reader and keyboard only users.
```

We will need to reflect this in our examples as well as our documentation... ☝️

**Comment by mathisscott** [comment]
@chlane 
Look good to you? ☝️

**Comment by chlane** [comment]
`<a href="#" role="region" aria-label="tooltip" class="tooltip tooltip-sm tooltip-top-right">
    <clr-icon shape="info-circle" size="24" title="Lorem ipsum sit"></clr-icon>
    <span id="user-gives-me-an-Id" class="tooltip-content" tabindex="-1" role="tooltip">Lorem ipsum sit</span>
</a>`

**Comment by chlane** [comment]
<a href="#" tabindex="-1" role="region" aria-label="tooltip" class="tooltip tooltip-sm tooltip-top-right">
    <clr-icon shape="info-circle" size="24" title="Lorem ipsum sit"></clr-icon>
    <span id="user-gives-me-an-Id" class="tooltip-content"  role="tooltip">Lorem ipsum sit</span>
</a>

**Comment by clane** [comment]
This issue is actually occurs with the Angular component version of the tooltip. See the updated steps to reproduce below. 

- Using VoiceOver/Safari, go to http://cx-guidelines.vmware.com/clarity-test-app2/
- Activate the 'Tooltip Demos' link
- Tab the tooltip icon under the "Angular Component" heading
- Observe that the text "Lorem" is visually displayed, but not announced by VoiceOver. The icon is instead announced as a 'group'.
